packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -559,20 +574,47 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
           handleRef(innerRef, listElement)
         }}
       >
-        <List
-          className={Dropdown.slotClassNames.itemsList}
-          {...accessibilityMenuProps}
-          styles={styles.list}
-          tabIndex={search ? undefined : -1} // needs to be focused when trigger button is activated.
-          aria-hidden={!open}
-          onFocus={this.handleTriggerButtonOrListFocus}
-          onBlur={this.handleListBlur}
-          items={open ? this.renderItems(styles, variables, getItemProps, highlightedIndex) : []}
-        />
+        {this.renderItemsList(
+          {
+            className: Dropdown.slotClassNames.itemsList,
+            ...accessibilityMenuProps,
+            styles: styles.list,
+            tabIndex: search ? undefined : -1, // needs to be focused when trigger button is activated.
+            'aria-hidden': !open,
+            onFocus: this.handleTriggerButtonOrListFocus,
+            onBlur: this.handleListBlur,
+            items: open ? this.renderItems(styles, variables, getItemProps, highlightedIndex) : [],
+          },
+          rtl,
+        )}
       </Ref>
     )
   }
 
+  private renderItemsList(listProps: ListProps, rtl: boolean): JSX.Element {
+    const { align, position, offset } = this.props
+
+    return (
+      <Positioner
+        align={align}
+        position={position}
+        offset={offset}
+        rtl={rtl}
+        target={this.selectedItemsRef}
+        children={popperChildrenProps => (
+          <UpdatableComponent
```

Comment:
I just wondered, why `UpdatableComponent` is not part of `Positioner`? For me it has much more sence

Reply:
I was thinking the same, but wanted to allow `Positioner` to receive components that don't necessarily update based on condition.

Alternative would be to:
- make `UpdatableComponent` HOC so that we don't need to pass `Component` and rest of props as props of `UpdatableComponent`
- move  `UpdatableComponent` to `Positioner` and add `updateDependencies` prop to `Positioner` as well and make it optional; then, if `updateDependencies` is:
  - `falsy`: we don't use `UpdatableComponent` at all
  - `[]`: component is rendered only when mounted (hooks behavior)
  - `[condition1, condition2, ..., conditionN]`: when at least a `conditionX` is true component rerenders (hooks behavior)

I think it makes sense. What do you think, @layershifter ?

Reply:
I suggest to remove `UpdatableComponent` at because it doesn't contain anything meaningful and place update logic in `Positioner`:

```tsx
  const scheduleUpdate = React.useRef<PopperChildrenProps['scheduleUpdate']>(null)
  
  React.useEffect(() => {
    if (scheduleUpdate.current) scheduleUpdate.current()
  }, updateDependencies) // `updateDependencies` is [] by default

  return (
    <Popper
      children={(childrenProps) => {
        scheduleUpdate.current = childrenProps.scheduleUpdate
        
        return children(childrenProps)
      }}
   />
```

Reply:
And may be `positioningDependencies` will be more accurate, not sure

Reply:
makes sense, the only thing missing was assigning the ref, added that following your suggestions, thanks!